### ISSUE TITLE
Problems with desktop environments, and HIDPI when using libdecor branch

### ISSUE THREAD

**Issue by yavko** [Admicos/minecraft-wayland#8]
The minecraft window has really weird scalling and 2 weird white window borders, full screen doesnt fix, and mouse cursor doesnt hide while in game. 
![fhgfhfhfghfgh](https://user-images.githubusercontent.com/15178513/138781345-3c441f54-f186-44ef-b47f-8f954199983d.png)
edit: manjaro gnome, 5.13 kernel, glfw-wayland-minecraft 3.3.4-2

**Comment by Prototik** [comment]
@yavko  I opened #9 with some fixes, related to Gnome, could you try it?

**Comment by yavko** [comment]
Ok @Prototik!

**Comment by yavko** [comment]
It built without issue! Thanks!

**Comment by yavko** [comment]
I wanted to add that the title bar is a bit weird it would be nice if it could use native GTK title bar!
![image](https://user-images.githubusercontent.com/15178513/139725803-ac105be6-5a4a-4fd2-b784-e52166ddb1fe.png)
and cursors don't hide and don't start from the center!
![image](https://user-images.githubusercontent.com/15178513/139727009-44f10fa1-33dc-459a-9e60-62ad7db21bc1.png)
And menus appear blurry

I don't know if the issue comes from this, but I hope there are fixes to these issues!

**Comment by Admicos** [comment]
In GNOME, you simply can't use the "Native GTK title bar" unless you use a majority of the GTK toolkit. (libdecor is, as far as I know, intended to be a "solution" to this in the future, but I am not exactly sure on that just yet) This is GNOME's decision (search for client-side decorations on their issue tracker) which is one of the reasons why I don't sue GNOME on my personal machines.

I'll take a look at the PR in a sec

**Comment by vchernin** [comment]
I noticed blurriness as well in https://github.com/Admicos/minecraft-wayland/issues/2, although unfortunately I never got around to fully testing and reporting that. If you can please forward that issue to the glfw libdecor PR as that’s fairly important for them to know.

**Comment by Admicos** [comment]
PR #11 should fix the mouse cursor issue, though I haven't tested it that much

**Comment by yavko** [comment]
> PR #11 should fix the mouse cursor issue, though I haven't tested it that much

ooooh
what about hidpi?

**Comment by yavko** [comment]
@Admicos  Are you on HIDPI?

**Comment by Admicos** [comment]
No, sorry. I don't have any hi-dpi display

**Comment by vchernin** [comment]
In theory for the sake of testing, you might be able to set even a 1080p display to 200% scaling temporarily (even though it will be janky, the test is still valid).

**Comment by yavko** [comment]
Yeah idk what the issue is since idk anything about opengl or window decorations :|

**Comment by vchernin** [comment]
I would guess either in the libdecor glfw patch or in libdecor itself, but I haven't been able to test this properly.

I know SDL has some sort of libdecor support, perhaps that would be a useful test.

**Comment by yavko** [comment]
btw i need to add something, i have only played on version 1.8.8 1.12.2 1.17.1 1.18 and only 1.17 and 1.18 have the non native window decor and ui bluriness

**Comment by vchernin** [comment]
That's strange, I will test it out in a moment.

**Comment by yavko** [comment]
> That's strange, I will test it out in a moment.

im using multimc btw

**Comment by vchernin** [comment]
There's possibly an incompatibility between older java that older minecraft versions use and modern glfw.

**Comment by yavko** [comment]
> There's possibly an incompatibility between older java that older minecraft versions use and modern glfw.

yeah im using graalvm 8 for everything below 1.17, graalvm 17 for 1.17 and above

**Comment by vchernin** [comment]
I assume that's a java fork? Yeah that's probably the reason. Not sure how minecraft/glfw upstream can workaround that problem.

**Comment by vchernin** [comment]
> https://github.com/Admicos/minecraft-wayland#requirements

**Comment by yavko** [comment]
> I assume that's a java fork? Yeah that's probably the reason, java 8 is very old. Not sure how minecraft/glfw upstream can workaround that problem.

its a java distro by oracle that is made on java instead of C and runs like wayyy faster

**Comment by yavko** [comment]
> 

oh lol i forgot

**Comment by vchernin** [comment]
[Reported here](https://github.com/glfw/glfw/pull/1693#issuecomment-991852655)

**Comment by yavko** [comment]
ok i see

**Comment by AlynxZhou** [comment]
I reproduced the border problem with AUR package, and it seems that resize works incorrectly.

**Comment by Admicos** [comment]
If you're on GNOME, manually build the `libdecoration` branch of the repo, optionally merging PR #11 and seeing if that fixes the problem it says it fixes.

Once the version of GLFW the libdecoration patch targets release officially (probably 3.4.x), then everything will be merged to `main` and uploaded to the AUR.

**Comment by yavko** [comment]
> If you're on GNOME, manually build the `libdecoration` branch of the repo, optionally merging PR #11 and seeing if that fixes the problem it says it fixes.
> 
> Once the version of GLFW the libdecoration patch targets release officially (probably 3.4.x), then everything will be merged to `main` and uploaded to the AUR.

i built the pr and it works great

**Comment by AlynxZhou** [comment]
> If you're on GNOME, manually build the `libdecoration` branch of the repo, optionally merging PR #11 and seeing if that fixes the problem it says it fixes.
> 
> Once the version of GLFW the libdecoration patch targets release officially (probably 3.4.x), then everything will be merged to `main` and uploaded to the AUR.

That works fine, but I have two monitors, when I press F11, it always fullscreen on my primary monitor.

**Comment by pajicadvance** [comment]
> If you're on GNOME, manually build the `libdecoration` branch of the repo, optionally merging PR #11 and seeing if that fixes the problem it says it fixes.
> 
> Once the version of GLFW the libdecoration patch targets release officially (probably 3.4.x), then everything will be merged to `main` and uploaded to the AUR.

This works great. The one small complaint I have is that the cursor remembers its last position when opening menus. Normally, the cursor should always be positioned in the center when doing so. Would it be possible to do this?

**Comment by yavko** [comment]
> > If you're on GNOME, manually build the `libdecoration` branch of the repo, optionally merging PR #11 and seeing if that fixes the problem it says it fixes.
> > Once the version of GLFW the libdecoration patch targets release officially (probably 3.4.x), then everything will be merged to `main` and uploaded to the AUR.
> 
> This works great. The one small complaint I have is that the cursor remembers its last position when opening menus. Normally, the cursor should always be positioned in the center when doing so. Would it be possible to do this?

you could make a macro ig

**Comment by Snivyer1910** [comment]
I have a weird problem where this happens to me... on KDE Plasma

**Comment by yavko** [comment]
> I have a weird problem where this happens to me... on KDE Plasma

Have you tried to build it using libdecor?

**Comment by Admicos** [comment]
> I have a weird problem where this happens to me... on KDE Plasma

Which issue? The weird decorations? HI-DPI blur? Mouse not hiding (#11)? All of them?

If the decorations are wrong on KDE, that's really weird. KDE Wayland should support server side decorations (they invented the protocol to do that AFAIK) so libdecor shouldn't be needed for it.

If it's one of the other issues, that's a bit more understandable, though I don't yet have a KDE environment to test in so I can't confirm anything yet.

**Comment by Snivyer1910** [comment]
it's the weird decoration (the gray bars), I didn't found any of the other issues

yeah I tried and now it works fine, except that when you close it instead of closing smoothly it crashes

I'mma try to play a bit and see if it crashes durning gameplay

**Comment by Admicos** [comment]
> yeah I tried and now it works fine, except that when you close it instead of closing smoothly it crashes (the JVM crashes, not the game)

I recently switched to KDE for my main system and I can also confirm this is happening.

Unfortunately, debugging this is outside my skill set so far, so I wouldn't expect a fast fix, but I'll see what I can do.
